_posts/2019-07-29-custom-domains-function-ingress.md

```diff
@@ -0,0 +1,235 @@
+---
+title: "Grab Custom Domains for your Functions with FunctionIngress"
+description: With OpenFaaS, you get a URL for every function by convention which is great and suits many people. This guide shows you have to get a Custom Domain for your functions using the new FunctionIngress concept.
+date: 2019-07-29
+image: /images/2019-custom-domains-function-ingress/1.jpg
+categories:
+  - community
+  - end-user
+  - ingress
+  - production
+  - kubernetes
+author_staff_member: alex
+dark_background: true
+---
+
+Find out how you can expand beyond the default URL given to each function with FunctionIngress.
+
+In this tutorial I'll show you how to use the new FunctionIngress CRD along with a new Kubernetes Operator to get Custom Domains for any OpenFaaS function deployed on Kubernetes.
+
+## Tutorial
+
+### How it works
+
+First of all, we install the new Kubernetes Operator called "IngressOperator". This Operator is responsible for translating CRD entries called FunctionIngress into two separate resources in the cluster: an `Ingress` definition and a TLS Certficate. The TLS certificate is optional, but highly recommended.
+
+### Pre-reqs
+
+* Kubernetes
+
+You need to have a Kubernetes cluster. I recommend using [DigitalOcean Kubernetes](https://www.digitalocean.com/) for an easy and cost-efficient set-up.
+
+* helm
+
+You'll need helm for most of the components we are using. If you are allergic to helm's server-side component called "tiller", then don't worry. You can simply use the "helm template" command to generate plain YAML files.
+
+[Install helm](https://github.com/openfaas/faas-netes/blob/master/HELM.md)
+
+* OpenFaaS via helm
+
+[Install OpenFaaS via helm](https://github.com/openfaas/faas-netes/blob/master/chart/openfaas/README.md#install)
+
+You should also install the OpenFaaS CLI using the instructions at [docs.openfaas.com](https://docs.openfaas.com/)
+
+* Nginx IngressController
+
+Traefik and Skipper are also supported, but we'll use Nginx for this tutorial. Install Nginx with the following:
+
+```
+helm install stable/nginx-ingress --name nginxingress --set rbac.create=true
+```
+
+* cert-manager (optional)
+
+Using TLS is optional, but highly recommended, so I will show you how to set it up.
+
+Use the [1.0 SSL for Kubernetes](https://docs.openfaas.com/reference/ssl/kubernetes-with-cert-manager/#10-ssl-for-the-gateway) documentation to set up cert-manager and an issuer. You can skip the steps for creating a `Certificate` and `Ingress` entry for the gateway.
```

Comment:
I think this should be re-written a little to better explain that the link is to the documentation guide for installing the cert-manager. I missed this entirely thinking the link was to the Kubernetes docs explaining SSL...

Perhaps something like: `Follow the [guide in the docs](link) for "Install cert-manager" and "Configure cert-manager". The other steps are not required for this tutorial`

That matches up with the titles in the linked docs making it easier to follow.